common/static/common/js/components/views/list.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Generic view to render a collection.
+ */
+;(function (define) {
+    'use strict';
+    define(['backbone', 'underscore'], function(Backbone, _) {
+        var ListView = Backbone.View.extend({
+            /**
+             * Override with the view used to render models in the collection.
+             */
+            itemViewClass: Backbone.View,
+
+            initialize: function (options) {
+                this.itemViewClass = options.itemViewClass || this.itemViewClass;
+                this.collection.on('add', this.render, this);
+                this.collection.on('remove', this.render, this);
+                this.collection.on('reset', this.render, this);
+                this.collection.on('sync', this.render, this);
+                this.collection.on('sort', this.render, this);
+                // Keep track of our children for garbage collection
+                this.itemViews = [];
+            },
+
+            /**
+             * Naive render.  Any time the collection changes at all we
+             * re-render the entire list due to ordering
```

Comment:
Good point. I think that when `render` is called as a callback for the above events. In those cases, `render` would be provided arguments for what changed. I like your idea of leaving `render` as a complete rendering. I will remove this comment, and assume the new `TODO` to be sufficient.